src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -148,6 +148,12 @@ struct ImplementationsFactory {
     bool has(impl_types impl_type) const;
 };
 
+struct BufferDescriptor {
```

Comment:
I think you can declare it before primitive_impl, so the forward declaration won't be needed